softirqs.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// Softirqs represents the softirq statistics.
+type Softirqs struct {
+	Hi      []uint64
+	Timer   []uint64
+	NetTx   []uint64
+	NetRx   []uint64
+	Block   []uint64
+	IrqPoll []uint64
```

Comment:
Please uppercase acronyms, see: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Sorry about the back and forth but I meant only uppercasing the acronyms, so
IRQPoll, HRTimer etc. I'd probably keep Tx and Rx since we call them Tx/Tx in other places as well.
In general, look at the other structs and how they are named, e.g:
- https://github.com/prometheus/procfs/blob/master/cpuinfo.go#L31
- https://github.com/prometheus/procfs/blob/116b5c4f80ab09a0a6a848a7606652821b90d065/net_ip_socket.go#L58

Reply:
no worries, I think that should be right now thank you.